### PR TITLE
Remove unused JS dependencies

### PR DESCRIPTION
**Pull request by gzdunek**
e counterpart https://github.com/gravitational/teleport.e/pull/4513

Before I start upgrading the package manager, I wanted to clean up the dependencies.
It turned out that many of the packages that we install are not used anywhere.

To check if a package is used, I looked for any imports of it in the repo.

I tested all the scripts from the root `package.json` that could be affected by this cleanup, all work.
Cleaning up the dependencies removed one warning (`react-test-renderer@18.3.1" has incorrect peer dependency "react@^18.3.1"`) from `yarn install` without adding more.

I also realized that we don't need most of the webpack stuff, Storybook config wants only a single function - `createDefaultConfig()`.

I'm going to backport this only to branch/v16, since it's not much different from master.
Best to review commit by commit.

### CODE REVIEW

File: web/packages/build/package.json [hunk 1]
```diff
@@ -54,7 +51,6 @@
     "@types/webpack": "^5.28.5",
     "@vitejs/plugin-react-swc": "^3.7.0",
     "babel-loader": "^8.2.5",
-    "core-js": "^3",
```
Comment (gzdunek) [inline web/packages/build/package.json:64]: I don't think a polyfill which is not imported anywhere, has any effect.
And we don't import `core-js`.

**Comment by gzdunek** [comment]
@ryanclark could you verify if TAG works fine with these changes? In the end I didn't remove any of deps that we pass to it, but I'd appreciate a double check.

BTW, after I removed `whatwg-fetch` library and switched to Node.js `fetch` the tests started to be flaky, so I left it as is.

File: web/packages/build/package.json [hunk 2]
```diff
@@ -98,10 +86,7 @@
     "vite": "^5.3.1",
     "vite-plugin-wasm": "^3.3.0",
     "vite-tsconfig-paths": "^4.3.2",
-    "webpack": "^5.76.2",
-    "webpack-bundle-analyzer": "^4.10.2",
-    "webpack-cli": "^4.10.0",
-    "webpack-dev-server": "^4.9.3"
+    "webpack": "^5.76.2"
```
Comment (ravicious) [inline web/packages/build/package.json:84]: Are we able to get rid of webpack too? `@storybook/builder-webpack5` already has it in its dependencies. We import it through TS comments in the config files, but at this point I think we can get rid of that.

**Review by ravicious — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]
Thanks for the cleanup

**Pull request merged**